[2025-07-25T13:58:25,758][ INFO] [test.context.support.DefaultTestContextBootstrapper]: Loaded default TestExecutionListener class names from location [META-INF/spring.factories]: [org.springframework.test.context.web.ServletTestExecutionListener, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener, org.springframework.test.context.support.DependencyInjectionTestExecutionListener, org.springframework.test.context.support.DirtiesContextTestExecutionListener, org.springframework.test.context.transaction.TransactionalTestExecutionListener, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener]
[2025-07-25T13:58:25,814][ INFO] [test.context.support.DefaultTestContextBootstrapper]: Using TestExecutionListeners: [org.springframework.test.context.web.ServletTestExecutionListener@24fb6a80, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener@48c35007, org.springframework.test.context.support.DependencyInjectionTestExecutionListener@72a85671, org.springframework.test.context.support.DirtiesContextTestExecutionListener@6722db6e, org.springframework.test.context.transaction.TransactionalTestExecutionListener@18f20260, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener@4ae33a11]
[2025-07-25T13:58:26,028][ INFO] [beans.factory.xml.XmlBeanDefinitionReader]: Loading XML bean definitions from class path resource [com/camp_us/context/root-context.xml]
[2025-07-25T13:58:26,265][ INFO] [beans.factory.xml.XmlBeanDefinitionReader]: Loading XML bean definitions from class path resource [com/camp_us/context/dataSource-context.xml]
[2025-07-25T13:58:26,322][ INFO] [beans.factory.xml.XmlBeanDefinitionReader]: Loading XML bean definitions from class path resource [com/camp_us/context/application-context.xml]
[2025-07-25T13:58:26,424][ INFO] [springframework.context.support.GenericApplicationContext]: Refreshing org.springframework.context.support.GenericApplicationContext@6d1d4d7: startup date [Fri Jul 25 13:58:26 KST 2025]; root of context hierarchy
[2025-07-25T13:58:26,644][ INFO] [springframework.context.support.PropertySourcesPlaceholderConfigurer]: Loading properties file from file [C:\mydisk\workspace\com.campUs.application\target\classes\com\camp_us\properties\dataSource.properties]
[2025-07-25T13:58:26,645][ INFO] [springframework.context.support.PropertySourcesPlaceholderConfigurer]: Loading properties file from file [C:\mydisk\workspace\com.campUs.application\target\classes\com\camp_us\properties\savedPath.properties]
[2025-07-25T13:58:26,658][ INFO] [beans.factory.annotation.AutowiredAnnotationBeanPostProcessor]: JSR-330 'javax.inject.Inject' annotation found and supported for autowiring
[2025-07-25T13:58:28,714][ INFO] [test.context.transaction.TransactionContext]: Began transaction (1) for test context [DefaultTestContext@e8e0dec testClass = TestMemberService, testInstance = com.campUs.application.TestMemberService@12d1f1d4, testMethod = testList@TestMemberService, testException = [null], mergedContextConfiguration = [MergedContextConfiguration@64b7225f testClass = TestMemberService, locations = '{classpath:com/camp_us/context/root-context.xml}', classes = '{}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{}', propertySourceProperties = '{}', contextCustomizers = set[[empty]], contextLoader = 'org.springframework.test.context.support.DelegatingSmartContextLoader', parent = [null]]]; transaction manager [org.springframework.jdbc.datasource.DataSourceTransactionManager@75fa1be3]; rollback [true]
[2025-07-25T13:58:29,000][ INFO] [test.context.transaction.TransactionContext]: Rolled back transaction for test context [DefaultTestContext@e8e0dec testClass = TestMemberService, testInstance = com.campUs.application.TestMemberService@12d1f1d4, testMethod = testList@TestMemberService, testException = [null], mergedContextConfiguration = [MergedContextConfiguration@64b7225f testClass = TestMemberService, locations = '{classpath:com/camp_us/context/root-context.xml}', classes = '{}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{}', propertySourceProperties = '{}', contextCustomizers = set[[empty]], contextLoader = 'org.springframework.test.context.support.DelegatingSmartContextLoader', parent = [null]]].
[2025-07-25T13:58:29,007][ INFO] [springframework.context.support.GenericApplicationContext]: Closing org.springframework.context.support.GenericApplicationContext@6d1d4d7: startup date [Fri Jul 25 13:58:26 KST 2025]; root of context hierarchy
[2025-07-25T13:59:12,045][ INFO] [test.context.support.DefaultTestContextBootstrapper]: Loaded default TestExecutionListener class names from location [META-INF/spring.factories]: [org.springframework.test.context.web.ServletTestExecutionListener, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener, org.springframework.test.context.support.DependencyInjectionTestExecutionListener, org.springframework.test.context.support.DirtiesContextTestExecutionListener, org.springframework.test.context.transaction.TransactionalTestExecutionListener, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener]
[2025-07-25T13:59:12,082][ INFO] [test.context.support.DefaultTestContextBootstrapper]: Using TestExecutionListeners: [org.springframework.test.context.web.ServletTestExecutionListener@3a94964, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener@5049d8b2, org.springframework.test.context.support.DependencyInjectionTestExecutionListener@6d0b5baf, org.springframework.test.context.support.DirtiesContextTestExecutionListener@631e06ab, org.springframework.test.context.transaction.TransactionalTestExecutionListener@2a3591c5, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener@34a75079]
[2025-07-25T13:59:12,314][ INFO] [beans.factory.xml.XmlBeanDefinitionReader]: Loading XML bean definitions from class path resource [com/camp_us/context/dataSource-context.xml]
[2025-07-25T13:59:12,487][ INFO] [springframework.context.support.GenericApplicationContext]: Refreshing org.springframework.context.support.GenericApplicationContext@2b58f754: startup date [Fri Jul 25 13:59:12 KST 2025]; root of context hierarchy
[2025-07-25T13:59:12,610][ INFO] [beans.factory.annotation.AutowiredAnnotationBeanPostProcessor]: JSR-330 'javax.inject.Inject' annotation found and supported for autowiring
[2025-07-25T13:59:13,187][ WARN] [springframework.test.context.TestContextManager]: Caught exception while allowing TestExecutionListener [org.springframework.test.context.transaction.TransactionalTestExecutionListener@2a3591c5] to process 'before' execution of test method [public void com.campUs.application.TestSqlSession.testSelectMemberByID()] for test instance [com.campUs.application.TestSqlSession@64b7225f]
org.springframework.transaction.CannotCreateTransactionException: Could not open JDBC Connection for transaction; nested exception is java.sql.SQLException: Cannot load JDBC driver class '${dataSource.driverClassName}'
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:245) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.getTransaction(AbstractPlatformTransactionManager.java:373) ~[spring-tx-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.transaction.TransactionContext.startTransaction(TransactionContext.java:98) ~[spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.transaction.TransactionalTestExecutionListener.beforeTestMethod(TransactionalTestExecutionListener.java:203) ~[spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.TestContextManager.beforeTestMethod(TestContextManager.java:269) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:74) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:252) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:94) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210) [.cp/:?]
Caused by: java.sql.SQLException: Cannot load JDBC driver class '${dataSource.driverClassName}'
	at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:54) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:459) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:525) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:731) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:204) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 25 more
Caused by: java.lang.ClassNotFoundException: ${dataSource.driverClassName}
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[?:?]
	at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
	at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:49) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:459) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:525) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:731) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:204) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 25 more
[2025-07-25T13:59:13,215][ WARN] [springframework.test.context.TestContextManager]: Caught exception while allowing TestExecutionListener [org.springframework.test.context.transaction.TransactionalTestExecutionListener@2a3591c5] to process 'before' execution of test method [public void com.campUs.application.TestSqlSession.testSelectMemberList()] for test instance [com.campUs.application.TestSqlSession@6a282fdd]
org.springframework.transaction.CannotCreateTransactionException: Could not open JDBC Connection for transaction; nested exception is java.sql.SQLException: Cannot load JDBC driver class '${dataSource.driverClassName}'
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:245) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.getTransaction(AbstractPlatformTransactionManager.java:373) ~[spring-tx-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.transaction.TransactionContext.startTransaction(TransactionContext.java:98) ~[spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.transaction.TransactionalTestExecutionListener.beforeTestMethod(TransactionalTestExecutionListener.java:203) ~[spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.TestContextManager.beforeTestMethod(TestContextManager.java:269) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:74) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:252) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:94) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210) [.cp/:?]
Caused by: java.sql.SQLException: Cannot load JDBC driver class '${dataSource.driverClassName}'
	at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:54) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:459) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:525) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:731) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:204) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 25 more
Caused by: java.lang.ClassNotFoundException: ${dataSource.driverClassName}
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[?:?]
	at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
	at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:49) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:459) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:525) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:731) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:204) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 25 more
[2025-07-25T13:59:13,230][ WARN] [springframework.test.context.TestContextManager]: Caught exception while allowing TestExecutionListener [org.springframework.test.context.transaction.TransactionalTestExecutionListener@2a3591c5] to process 'before' execution of test method [public void com.campUs.application.TestSqlSession.testUpdateMember()] for test instance [com.campUs.application.TestSqlSession@10c2064a]
org.springframework.transaction.CannotCreateTransactionException: Could not open JDBC Connection for transaction; nested exception is java.sql.SQLException: Cannot load JDBC driver class '${dataSource.driverClassName}'
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:245) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.getTransaction(AbstractPlatformTransactionManager.java:373) ~[spring-tx-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.transaction.TransactionContext.startTransaction(TransactionContext.java:98) ~[spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.transaction.TransactionalTestExecutionListener.beforeTestMethod(TransactionalTestExecutionListener.java:203) ~[spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.TestContextManager.beforeTestMethod(TestContextManager.java:269) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:74) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:252) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:94) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210) [.cp/:?]
Caused by: java.sql.SQLException: Cannot load JDBC driver class '${dataSource.driverClassName}'
	at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:54) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:459) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:525) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:731) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:204) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 25 more
Caused by: java.lang.ClassNotFoundException: ${dataSource.driverClassName}
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[?:?]
	at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
	at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:49) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:459) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:525) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:731) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:204) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 25 more
[2025-07-25T13:59:13,242][ WARN] [springframework.test.context.TestContextManager]: Caught exception while allowing TestExecutionListener [org.springframework.test.context.transaction.TransactionalTestExecutionListener@2a3591c5] to process 'before' execution of test method [public void com.campUs.application.TestSqlSession.testDeleteMember()] for test instance [com.campUs.application.TestSqlSession@28fd3dc1]
org.springframework.transaction.CannotCreateTransactionException: Could not open JDBC Connection for transaction; nested exception is java.sql.SQLException: Cannot load JDBC driver class '${dataSource.driverClassName}'
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:245) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.getTransaction(AbstractPlatformTransactionManager.java:373) ~[spring-tx-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.transaction.TransactionContext.startTransaction(TransactionContext.java:98) ~[spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.transaction.TransactionalTestExecutionListener.beforeTestMethod(TransactionalTestExecutionListener.java:203) ~[spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.TestContextManager.beforeTestMethod(TestContextManager.java:269) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:74) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:252) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:94) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210) [.cp/:?]
Caused by: java.sql.SQLException: Cannot load JDBC driver class '${dataSource.driverClassName}'
	at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:54) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:459) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:525) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:731) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:204) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 25 more
Caused by: java.lang.ClassNotFoundException: ${dataSource.driverClassName}
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[?:?]
	at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
	at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:49) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:459) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:525) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:731) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:204) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 25 more
[2025-07-25T13:59:13,258][ WARN] [springframework.test.context.TestContextManager]: Caught exception while allowing TestExecutionListener [org.springframework.test.context.transaction.TransactionalTestExecutionListener@2a3591c5] to process 'before' execution of test method [public void com.campUs.application.TestSqlSession.testInsertMember()] for test instance [com.campUs.application.TestSqlSession@15e0fe05]
org.springframework.transaction.CannotCreateTransactionException: Could not open JDBC Connection for transaction; nested exception is java.sql.SQLException: Cannot load JDBC driver class '${dataSource.driverClassName}'
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:245) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.getTransaction(AbstractPlatformTransactionManager.java:373) ~[spring-tx-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.transaction.TransactionContext.startTransaction(TransactionContext.java:98) ~[spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.transaction.TransactionalTestExecutionListener.beforeTestMethod(TransactionalTestExecutionListener.java:203) ~[spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.TestContextManager.beforeTestMethod(TestContextManager.java:269) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:74) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:252) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:94) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) [junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191) [spring-test-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210) [.cp/:?]
Caused by: java.sql.SQLException: Cannot load JDBC driver class '${dataSource.driverClassName}'
	at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:54) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:459) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:525) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:731) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:204) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 25 more
Caused by: java.lang.ClassNotFoundException: ${dataSource.driverClassName}
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[?:?]
	at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
	at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:49) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:459) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:525) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:731) ~[commons-dbcp2-2.9.0.jar:2.9.0]
	at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:204) ~[spring-jdbc-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 25 more
[2025-07-25T13:59:13,263][ INFO] [springframework.context.support.GenericApplicationContext]: Closing org.springframework.context.support.GenericApplicationContext@2b58f754: startup date [Fri Jul 25 13:59:12 KST 2025]; root of context hierarchy
[2025-07-25T13:59:34,944][ INFO] [test.context.support.DefaultTestContextBootstrapper]: Loaded default TestExecutionListener class names from location [META-INF/spring.factories]: [org.springframework.test.context.web.ServletTestExecutionListener, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener, org.springframework.test.context.support.DependencyInjectionTestExecutionListener, org.springframework.test.context.support.DirtiesContextTestExecutionListener, org.springframework.test.context.transaction.TransactionalTestExecutionListener, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener]
[2025-07-25T13:59:34,980][ INFO] [test.context.support.DefaultTestContextBootstrapper]: Using TestExecutionListeners: [org.springframework.test.context.web.ServletTestExecutionListener@24fb6a80, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener@48c35007, org.springframework.test.context.support.DependencyInjectionTestExecutionListener@72a85671, org.springframework.test.context.support.DirtiesContextTestExecutionListener@6722db6e, org.springframework.test.context.transaction.TransactionalTestExecutionListener@18f20260, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener@4ae33a11]
[2025-07-25T13:59:35,162][ INFO] [beans.factory.xml.XmlBeanDefinitionReader]: Loading XML bean definitions from class path resource [com/camp_us/context/root-context.xml]
[2025-07-25T13:59:35,332][ INFO] [beans.factory.xml.XmlBeanDefinitionReader]: Loading XML bean definitions from class path resource [com/camp_us/context/dataSource-context.xml]
[2025-07-25T13:59:35,378][ INFO] [beans.factory.xml.XmlBeanDefinitionReader]: Loading XML bean definitions from class path resource [com/camp_us/context/application-context.xml]
[2025-07-25T13:59:35,444][ INFO] [springframework.context.support.GenericApplicationContext]: Refreshing org.springframework.context.support.GenericApplicationContext@6d1d4d7: startup date [Fri Jul 25 13:59:35 KST 2025]; root of context hierarchy
[2025-07-25T13:59:35,636][ INFO] [springframework.context.support.PropertySourcesPlaceholderConfigurer]: Loading properties file from file [C:\mydisk\workspace\com.campUs.application\target\classes\com\camp_us\properties\dataSource.properties]
[2025-07-25T13:59:35,637][ INFO] [springframework.context.support.PropertySourcesPlaceholderConfigurer]: Loading properties file from file [C:\mydisk\workspace\com.campUs.application\target\classes\com\camp_us\properties\savedPath.properties]
[2025-07-25T13:59:35,644][ INFO] [beans.factory.annotation.AutowiredAnnotationBeanPostProcessor]: JSR-330 'javax.inject.Inject' annotation found and supported for autowiring
[2025-07-25T13:59:37,350][ INFO] [test.context.transaction.TransactionContext]: Began transaction (1) for test context [DefaultTestContext@f679798 testClass = TestMemberService, testInstance = com.campUs.application.TestMemberService@1a6cf771, testMethod = testList@TestMemberService, testException = [null], mergedContextConfiguration = [MergedContextConfiguration@7f92b990 testClass = TestMemberService, locations = '{classpath:com/camp_us/context/root-context.xml}', classes = '{}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{}', propertySourceProperties = '{}', contextCustomizers = set[[empty]], contextLoader = 'org.springframework.test.context.support.DelegatingSmartContextLoader', parent = [null]]]; transaction manager [org.springframework.jdbc.datasource.DataSourceTransactionManager@59d77850]; rollback [true]
[2025-07-25T13:59:37,495][ INFO] [test.context.transaction.TransactionContext]: Rolled back transaction for test context [DefaultTestContext@f679798 testClass = TestMemberService, testInstance = com.campUs.application.TestMemberService@1a6cf771, testMethod = testList@TestMemberService, testException = [null], mergedContextConfiguration = [MergedContextConfiguration@7f92b990 testClass = TestMemberService, locations = '{classpath:com/camp_us/context/root-context.xml}', classes = '{}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{}', propertySourceProperties = '{}', contextCustomizers = set[[empty]], contextLoader = 'org.springframework.test.context.support.DelegatingSmartContextLoader', parent = [null]]].
[2025-07-25T13:59:37,502][ INFO] [springframework.context.support.GenericApplicationContext]: Closing org.springframework.context.support.GenericApplicationContext@6d1d4d7: startup date [Fri Jul 25 13:59:35 KST 2025]; root of context hierarchy
